src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -126,18 +129,26 @@ void fillInnerParts() {
     }
 
     /**
-     * Change the view to show Person.
+     * Change the view to show the Person list.
      */
     void fillPerson() {
-        personListPanel = new PersonListPanel(logic.getFilteredPersonList());
+        entryListPanelPlaceholder.getChildren().clear();
         entryListPanelPlaceholder.getChildren().add(personListPanel.getRoot());
     }
 
     /**
-     * Change the view to show Event
+     * Change the view to show the Company list.
+     */
+    void fillCompany() {
+        entryListPanelPlaceholder.getChildren().clear();
+        entryListPanelPlaceholder.getChildren().add(companyListPanel.getRoot());
+    }
+
+    /**
+     * Change the view to show the Event list.
      */
     void fillEvent() {
-        eventListPanel = new EventListPanel(logic.getFilteredEventList());
+        entryListPanelPlaceholder.getChildren().clear();
         entryListPanelPlaceholder.getChildren().add(eventListPanel.getRoot());
     }
 
```

Comment:
This comment is for 3 functions : fillPerson(), fillCompany(), and fillEvent()
Correct me if I'm wrong, but I think that these lines should not be erased?

- personListPanel = new PersonListPanel(logic.getFilteredPersonList());
- eventListPanel = new EventListPanel(logic.getFilteredEventList());
- companyListPanel = new CompanyListPanel(logic.getFilteredCompanyList());

This is because those 3 lines will refresh the events, companies, or persons that will be displayed. These will be useful when implementing the find command for the companies and events as we will just call this to refresh the display accordingly

Reply:
From my understanding, the events/companies/person lists displayed will automatically refresh even without those lines. You can see this by calling the `find` command in this PR. I'm not entirely sure why this happens (seems to be a Java FX feature?) but it does mean we don't need to create a new object every time.